### ISSUE TITLE
The @key decorator appears to be ignored by the openapi3 emitter

### ISSUE THREAD

**Issue by mikekistler** [microsoft/cadl#1394]
It seems that the `@key` (built-in) decorator has no effect on the output of the openapi3 emitter. This does appear to be consistent with the current documentation -- `@key` is not mentioned at all in the openapi3 emitter doc.  But it is not what I expected or what some users might expect.

What I expected was that if I have a model with a property decorated with `@key` then if I pass that model as an operation parameter the @key property would be appended to the path.  E.g. I was expecting
```
model Widget {
  @key id: string;
  weight: int32;
}

@route("/widgets")
@put op createOrUpdate(...Widget): Widget | Error;
```

to produce a `put` operation on "/widgets/{id}".  But it does not.  I tried changing the spread of Widget to declaring it as a body parameter, but this gives the same result. Note: the cadl-autorest emitter behaves the same way.

Note: If I change "@key" to "@path" then I do get `{id}` appended to the path.

Perhaps my expectation was just wrong on this, and that would be fine. But if that's the case, then I recommend that we remove the `@key` decorator from `Widget.id` in the "Http Service" example of the public playground. Or maybe replace it with `@path`.

**Comment by daviwil** [comment]
`@key` is just metadata on a model property, it has no inherent representation for either of the OpenAPI emitters.  The way we use it in practice is via the `KeysOf<T>` family of wrapper model types.  Those types will filter out all properties of `T` that aren't `key` and then add the `path` decorator to them.  To fix your example, you could do this:

```
import "@cadl-lang/rest";

using Cadl.Http;
using Cadl.Rest;

model Widget {
  @key id: string;
  weight: int32;
}

@error
model Error {
  code: string;
}

@route("/widgets")
@put op createOrUpdate(...Cadl.Rest.Resource.KeysOf<Widget>): Widget | Error;
```

[Playground Link](https://cadlplayground.z22.web.core.windows.net/?c=aW1wb3J0ICJAY2FkbC1sYW5nL3Jlc3QiOwoKdXNpbmcgQ2FkbC5IdHRwO8wRUmVzdDsKCm1vZGVsIFdpZGdldCB7CiAgQGtleSBpZDogc3RyaW5nOwogIHdlaWdodDogaW50MzI7Cn0KCkBlcnJvcsc9RcQMxTxjb2RlyjnEKHJvdXRlKCIvd8VhcyIpCkBwdXQgb3AgY3JlYXRlT3JVcGRhdGUoLi7kAJJvdXJjZS5LZXlzT2Y85gCVPik66ACffMZrOw%3D%3D)

**Comment by markcowl** [comment]
Fixing in http sample

**Comment by mikekistler** [comment]
Fixed in #1429
